### PR TITLE
GraphDoc_ActiveInactiveAssignments

### PR DESCRIPTION
**Pull request by AshwaniBansal1**
Documentation for Active-InactiveAssignments

### CODE REVIEW

File: api-reference/beta/api/educationassignment-get.md [hunk 1]
```diff
@@ -153,6 +156,155 @@ Content-length: 279
...
+#### Response
+The following is an example of the response with status as inactive, when `Prefer: include-unknown-enum-members` is provided in the request header.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.educationAssignment",
+  "isCollection": true
```
Comment (cristobal-buenrostro) [inline api-reference/beta/api/educationassignment-get.md:181]: remove this line, it is indicating that the response will return a collection, that's the reason it's expecting a "value" property.

File: api-reference/beta/api/educationassignment-get.md [hunk 2]
```diff
@@ -153,6 +156,155 @@ Content-length: 279
...
+#### Response
+The following is an example of the response when Prefer: include-unknown-enum-members is not provided in the request header.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.educationAssignment",
+  "isCollection": true
```
Comment (cristobal-buenrostro) [inline api-reference/beta/api/educationassignment-get.md:255]: same here, remove this line

File: api-reference/beta/api/educationassignment-activate.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+### Response
+The following is an example of the response.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.educationAssignment",
+  "isCollection": true
```
Comment (cristobal-buenrostro) [inline api-reference/beta/api/educationassignment-activate.md:74]: remove

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+#### Response
+The following is an example of the response when `Prefer: include-unknown-enum-members` is provided in the request header.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.educationAssignment",
+  "isCollection": true
```
Comment (cristobal-buenrostro) [inline api-reference/beta/api/educationassignment-deactivate.md:151]: remove this line

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+#### Response
+The following is an example of the response when Prefer: include-unknown-enum-members is NOT provided in the request header.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.educationAssignment",
+  "isCollection": true
```
Comment (cristobal-buenrostro) [inline api-reference/beta/api/educationassignment-deactivate.md:75]: remove

**Review by cristobal-buenrostro — APPROVED** [review]

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
+---
+title: "Deactive educationAssignment"
```
Comment (richardle798) [inline api-reference/beta/api/educationassignment-deactivate.md:2]: Deactivate

File: api-reference/beta/resources/educationassignment.md [hunk 7]
```diff
@@ -42,6 +42,8 @@ Inherits from [entity](../resources/entity.md).
 |[Attach rubric](../api/educationassignment-put-rubric.md)|None|Attach an existing **educationRubric** to this **assignment**.|
 |[Remove rubric](../api/educationassignment-delete-rubric.md)|None|Detach the **educationRubric** from this **assignment**.|
 |[Get delta](../api/educationassignment-delta.md)|[educationAssignment](../resources/educationassignment.md) collection|Get a list of newly created or updated **assignments** without having to perform a full read of the collection.|
+|[Activate assignment](../api/educationassignment-activate.md) |[educationAssignment](educationassignment.md) | Activate an inactive **educationAssignment** to signal that the assignment has action items for teachers and students.|
+|[Activate assignment](../api/educationassignment-deactivate.md) |[educationAssignment](educationassignment.md) | Mark an assigned **educationAssignment** as inactive to signal that the assignment has no further action items for both teachers and students.|
```
Comment (richardle798) [inline api-reference/beta/resources/educationassignment.md:46]: Deactivate assignment

**Review by richardle798 — CHANGES_REQUESTED** [review]

**Review by richardle798 — APPROVED** [review]

**Comment by dodaromike** [comment]
Lines 18,26 in the changelog need to be fixed.

**Comment by dodaromike** [comment]
@cristobal-buenrostro Broken links need to be fixed at changelog/Microsoft.EducationAssignment.json: Line(s) 18,26.  @AshwaniBansal1 says you are now working on this PR.  I've tried fix the links but failed.

**Review by dodaromike — APPROVED** [review]

**Review by dodaromike — CHANGES_REQUESTED** [review]
@cristobal-buenrostro Broken links need to be fixed at changelog/Microsoft.EducationAssignment.json: Line(s) 18,26. @AshwaniBansal1 says you are now working on this PR. I've tried fix the links but failed.

**Comment by cristobal-buenrostro** [comment]
> @cristobal-buenrostro Broken links need to be fixed at changelog/Microsoft.EducationAssignment.json: Line(s) 18,26. @AshwaniBansal1 says you are now working on this PR. I've tried fix the links but failed.

@dodaromike , links are fixed now.

**Comment by dodaromike** [comment]
Thanks @cristobal-buenrostro for link fixes.  The new problems don't appear to be caused by the current PR.  I'll approve it.

File: api-reference/beta/api/educationassignment-activate.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+| Authorization | Bearer {token}. Required.   |
+| Content-Type  | application/json. Required. |
+
+## Request body
+
+Do not supply a request body for this method.
+
+## Response
+
+If successful, this method returns a `200 OK` response code and a [status](/graph/api/resources/educationassignment?#properties) value of `assigned` in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-activate.md:47]: To be clear, from the example, it looks like and educatioAssignment object is returned in the response body, and the status property has a value of `assigned`. Please clarify what is returned in the response.

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+<!-- { "blockType": "ignored" } -->
+```http
+POST /education/classes/{classId}/assignments/{assignmentId}/deactivate
+```
+
+## Request headers
+| Header        | Value                       |
+| :------------ | :------------------------   |
+| Authorization | Bearer {token}. Required.   |
+| Content-Type  | application/json. Required. |
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-deactivate.md:39]: This header isn't included when there isn't a request body. Also, this section doesn't list the optional Prefer header described in the examples. Please make sure that the Request headers section is accurate.

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+Do not supply a request body for this method.
+
+## Response
+
+If successful, this method returns a `200 OK` response code and a [status](/graph/api/resources/educationassignment?#properties) `inactive` in the response body.
+
+## Examples
+
+### Example 1: Mark assignment inactive-request without optional Prefer header
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-deactivate.md:51]: Can you explain why they would and would not include this request header?

File: api-reference/beta/api/educationassignment-activate.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+<!-- { "blockType": "ignored" } -->
+```http
+POST /education/classes/{classId}/assignments/{assignmentId}/activate
+```
+
+## Request headers
+| Header        | Value                       |
+| :------------ | :------------------------   |
+| Authorization | Bearer {token}. Required.   |
+| Content-Type  | application/json. Required. |
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-activate.md:39]: We should remove this header if it's not used in the request.

File: api-reference/beta/api/educationassignment-get.md [hunk 12]
```diff
@@ -153,6 +156,153 @@ Content-length: 279
 }
 ```
 
+## Example 2: Get assignment in inactive state with optional Prefer header
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-get.md:159]: Please update the Request headers section and add context about using this optional header. Also, this should be an H3, not an H2.

File: api-reference/beta/api/educationassignment-activate.md (hunk 8 above)
Comment (cristobal-buenrostro) [reply api-reference/beta/api/educationassignment-activate.md:47]: updated

File: api-reference/beta/api/educationassignment-activate.md (hunk 11 above)
Comment (cristobal-buenrostro) [reply api-reference/beta/api/educationassignment-activate.md:39]: added in the request

File: api-reference/beta/api/educationassignment-deactivate.md (hunk 9 above)
Comment (cristobal-buenrostro) [reply api-reference/beta/api/educationassignment-deactivate.md:39]: added content type in the request body.
And added prefer as optional.

File: api-reference/beta/api/educationassignment-deactivate.md (hunk 10 above)
Comment (cristobal-buenrostro) [reply api-reference/beta/api/educationassignment-deactivate.md:51]: done

File: api-reference/beta/api/educationassignment-get.md (hunk 12 above)
Comment (cristobal-buenrostro) [reply api-reference/beta/api/educationassignment-get.md:159]: updated to H3, more context added in the description about the optional parameter.

**Review by Lauragra — APPROVED** [review]

**Comment by dodaromike** [comment]
@cristobal-buenrostro  Please set this PR "Ready to merge".

File: api-reference/beta/api/educationassignment-activate.md (hunk 11 above)
Comment (Lauragra) [reply api-reference/beta/api/educationassignment-activate.md:39]: Why is the header used in the request when the request doesn't have a JSON body?

File: api-reference/beta/api/educationassignment-deactivate.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+The following is an example of the request.
+
+<!-- {
+  "blockType": "request",
+  "sampleKeys": ["ffac078e-1b63-42d0-bc2a-d280896e289a","2b8090d7-8de9-4fb4-af5d-2e2f68ae098a"],
+  "name": "post_deactivateAssignment_withoutheader"
+}-->
+```http
+POST https://graph.microsoft.com/beta/education/classes/ffac078e-1b63-42d0-bc2a-d280896e289a/assignments/2b8090d7-8de9-4fb4-af5d-2e2f68ae098a/deactivate
+Content-type: application/json
```
Comment (Lauragra) [inline api-reference/beta/api/educationassignment-deactivate.md:65]: Same question here - there is no request body.

**Comment by Lauragra** [comment]
@AshwaniBansal1 , this PR has API Doctor errors that appear to be related to the inclusion of the request headers. Can you please fix and make sure that the PR passes validation before we publish? Thanks.

**Comment by dodaromike** [comment]
> @AshwaniBansal1 , this PR has API Doctor errors that appear to be related to the inclusion of the request headers. Can you please fix and make sure that the PR passes validation before we publish? Thanks.

Including @cristobal-buenrostro

**Pull request merged**